client/components/signup-form/index.jsx

```diff
@@ -476,4 +478,9 @@ class SignupForm extends Component {
 	}
 }
 
-export default localize( SignupForm );
+export default connect(
+	null,
+	dispatch => ( {
+		trackLoginMidFlow: () => dispatch( recordTracksEvent( 'calypso_signup_login_midflow' ) )
```

Comment:
Regarding the name of this event, I'm worried that this step doesn't (or won't) always happen in the middle of the flow. It also doesn't convey what's happening - we usually use [a verb](https://github.com/Automattic/wp-calypso/tree/master/client/lib/analytics#tracks-naming-conventions) to denote that. Since we already have a `calypso_signup_goto_invalid_step` event, what about using `calypso_signup_goto_login_step` here?